main.py

```diff
@@ -64,3 +65,21 @@ def handle_service_exception(
     return JSONResponse(
         status_code=status.HTTP_400_BAD_REQUEST, content={"detail": str(exception)}
     )
+
+
+@app.exception_handler(AuthenticationException)
+def handle_authentication_exception(
+    request: Request, exception: AuthenticationException
+) -> JSONResponse:
+    return JSONResponse(
+        status_code=status.HTTP_400_BAD_REQUEST, content={"detail": str(exception)}
+    )
+
+
+@app.exception_handler(AuthorizationException)
+def handle_authorization_exception(
+    request: Request, exception: AuthorizationException
+) -> JSONResponse:
+    return JSONResponse(
+        status_code=status.HTTP_403_FORBIDDEN, content={"detail": str(exception)}
+    )
```

Comment:
Change status code to 404, because suggesting the user that resource is unavilable for them but still exists may become a vulnerability. In case of this app, all resources are owner by some user so 404 code may be a good security solution